### DISCUSSION TITLE
Are the new assistants API's supported?

### DISCUSSION THREAD

**Discussion by obsidianbtc** [mgallo/openai.ex#56] [General]
Was curious if anyone is currently working on integrating https://platform.openai.com/docs/assistants/overview

**Comment by ArthV** [comment]
I was going to ask the exact same question, it would be helpful to know if an effort will start in this direction

**Comment by ArthV** [comment]
@mgallo Thanks for this new update, it's really helpful. I wanted to know if you were planning to add the `submit_tool_outputs` endpoint?

**Reply by mgallo** [reply]
Hi! Thanks for reporting that, I had not checked properly for the presence of all APIs support for threads, sorry.
btw, the new version [v0.6.1](https://hex.pm/packages/openai/0.6.1) supports `submit_tool_outputs `. Unfortunately, I could not test on real cases, because my test `runs` were not in the required state, but it should work properly, Let me know in case

**Reply by ArthV** [reply]
Hi! I checked on my local and everything works fine: 

Here the request 
```
{url,{hackney_url,hackney_ssl,https,<<"api.openai.com">>,
<<"/v1/threads/thread_id/runs/run_id/submit_tool_outputs">>,
<<"/v1/threads/thread_id/runs/run_id/submit_tool_outputs">>,
<<>>,<<>>,"api.openai.com",443,<<>>,<<>>}},
{headers,[{<<"OpenAI-Beta">>,<<"assistants=v1">>},
{<<"OpenAI-Organization">>,
<<"org-id">>},
{<<"Authorization">>,
<<"Bearer openai_API_KEY »>>},
{<<"Content-type">>,<<"application/json">>}]},
{body,<<"{"tool_outputs":[{"output":"TRUE","params":"{\"first_element\":3,\"operator\":\"+\",\"result\":7,\"second_element\":4}","tool_call_id":"call_id"}]}">>},
``` 

And the body of the answer I received

```
 <<"{\n  \"id\": \"run_id\",\n  \"object\": \"thread.run\",\n  \"created_at\": 1700562696,\n  \"assistant_id\": \"asst_id\",\n  \"thread_id\": \"thread_id\",\n  \"status\": \"queued\",\n  \"started_at\": 1700562696,\n  \"expires_at\": 1700563296,\n  \"cancelled_at\": null,\n  \"failed_at\": null,\n  \"completed_at\": null,\n  \"last_error\": null,\n  \"model\": \"gpt-4-1106-preview\",\n  \"instructions\": \"You act as a web scraper\",\n  \"tools\": [\n    {\n      \"type\": \"retrieval\"\n    },\n    {\n  ">>,
```

**Reply by mgallo** [reply]
awesome! Thanks for testing it 👍
